### DISCUSSION TITLE
WS2812 LEDs Display Incorrect Colors When WiFi is Connected

### DISCUSSION THREAD

**Discussion by mnameghi** [Makuna/NeoPixelBus#812] [Q&A] [answered]
Hi,

I used an ESP-01 with two WS2812 LEDs. I connected the LEDs to GPIO2, and everything works fine when the WiFi is not connected. However, when I connect to WiFi, the LEDs display the wrong colors. For example, I intend for the LEDs to be red, but they turn on with different colors instead.

I have tried using the asynchronous UART method, but the result was the same as with the standard UART method.

**Comment by Makuna** [comment, marked as answer]
The discussion area is the place to ask questions.

Try the DMA method with the required pin 3 to confirm it works for you.
https://github.com/Makuna/NeoPixelBus/wiki/ESP8266-NeoMethods
Do you have a level shifter between the ESP and the LEDs?

I can't control WiFi and it has a higher priority than anything I can do.  This is why I recommended the DMA method as it should not be affected by WiFi at all.

If you test with the DMA method and it still has issues, then I would suspect a signal issue around power, either the signal level shifter is not fast enough or there is noise induced on the data pin (caps and resistors can help if you use an osciliscope to check and tune).
Lastly, a power issue where not enough power is provided to the ESP. When it communicates on WiFi it draws upto 800mA.  This may cause voltage dip and then cause the data signal to drop below what the LEDs can read (assuming no level shifter).
